webapp/channels/src/components/admin_console/access_control/policy_details/policy_details.tsx

```diff
@@ -0,0 +1,412 @@
+import TextSetting from 'components/widgets/settings/text_setting';
+
+import {getHistory} from 'utils/browser_history';
+
+import './policy_details.scss';
+
+const DEFAULT_PAGE_SIZE = 10;
+
+interface PolicyActions {
+    fetchPolicy: (id: string) => Promise<ActionResult>;
+    createPolicy: (policy: AccessControlPolicy) => Promise<ActionResult>;
+    deletePolicy: (id: string) => Promise<ActionResult>;
+    searchChannels: (id: string, term: string, opts: ChannelSearchOpts) => Promise<ActionResult>;
+    setNavigationBlocked: (blocked: boolean) => void;
+    assignChannelsToAccessControlPolicy: (policyId: string, channelIds: string[]) => Promise<ActionResult>;
+    unassignChannelsFromAccessControlPolicy: (policyId: string, channelIds: string[]) => Promise<ActionResult>;
+    getAccessControlExpressionAutocomplete: () => Promise<ActionResult>;
+}
+
+export interface PolicyDetailsProps {
+    policyId?: string;
+    actions: PolicyActions;
+    autocompleteResult: AccessControlExpressionAutocomplete;
+}
+
+interface ChannelChanges {
+    removed: Record<string, ChannelWithTeamData>;
+    added: Record<string, ChannelWithTeamData>;
+    removedCount: number;
+}
+
+const PolicyDetails: React.FC<PolicyDetailsProps> = ({policyId, actions}) => {
+    const [policyName, setPolicyName] = useState('');
+    const [expression, setExpression] = useState('');
+    const [autoSyncMembership, setAutoSyncMembership] = useState(false);
+    const [serverError, setServerError] = useState(false);
+    const [addChannelOpen, setAddChannelOpen] = useState(false);
+    const [editorMode, setEditorMode] = useState<'cel' | 'table'>('cel');
+    const [channelChanges, setChannelChanges] = useState<ChannelChanges>({
+        removed: {},
+        added: {},
+        removedCount: 0,
+    });
+    const [saveNeeded, setSaveNeeded] = useState(false);
+    const [channelsCount, setChannelsCount] = useState(0);
+    const [autocompleteResult, setAutocompleteResult] = useState<AccessControlExpressionAutocomplete>({entities: {}});
+
+    useEffect(() => {
+        loadPage();
+    }, [policyId]);
+
+    const loadPage = async () => {
+        if (policyId) {
+            const result = await actions.fetchPolicy(policyId);
+
+            // Set policy name and expression after fetching the policy
+            setPolicyName(result.data?.name || '');
+            setExpression(result.data?.rules?.[0]?.expression || '');
+
+            // Search for channels after setting the policy details
+            const channelsResult = await actions.searchChannels(policyId, '', {per_page: DEFAULT_PAGE_SIZE});
+            setChannelsCount(channelsResult.data?.total_count || 0);
+
+            const autocompleteResult = await actions.getAccessControlExpressionAutocomplete();
+            if (autocompleteResult.data) {
+                setAutocompleteResult(autocompleteResult.data);
+            }
+        }
+    };
+
+    const handleSubmit = async () => {
+        try {
```

Comment:
Wrapping everything under try-catch is an anti-pattern. Can you instead have only the specific code in try catch?

Reply:
Sure, I'll group them logically.